### PR TITLE
miscellaneous crash fixes for windows 64 and osx builds.

### PR DESCRIPTION
**Pull request by dbadb**
Without this, the example file, osc_dump.ck crashes when listening
on OSC messages. (windows 64bit release build).

### CODE REVIEW

**Review by gewang — APPROVED** [review]
thank you!

**Comment by gewang** [comment]
Thank you for this! (there were some conflicts with recent updates, so a few of these may be rolled back.)

File: src/core/makefile.x/makefile.osx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # uncomment the following to force 32-bit compilation
 # FORCE_M32=-m32
 
-ifneq ($(shell sw_vers -productVersion | egrep '10\.(6|7|8|9|10|11|12|13|14|15)(\.[0-9]+)?'),)
+# the sw_vers trick only works after 10.6
```
Comment (gewang) [inline src/core/makefile.x/makefile.osx]: recently pushed a different and conflicting change here and thus this change has been not include for now. if you'd like, please take a look in main branch and advise how to proceed. thanks!

Comment (dbadb) [reply src/core/makefile.x/makefile.osx]: both fixes seem equivalent. no problem with the different fix from my POV.

**Pull request merged**